local-cluster/tests/local_cluster.rs

```diff
@@ -3074,19 +3074,18 @@ fn run_test_load_program_accounts(scan_commitment: CommitmentConfig) {
 #[test]
 #[serial]
 fn test_no_optimistic_confirmation_violation_with_tower() {
-    do_test_optimistic_confirmation_violation_with_or_without_tower(true);
+    do_test_lockout_violation_with_or_without_tower(true);
 }
 
 #[test]
 #[serial]
 fn test_optimistic_confirmation_violation_without_tower() {
-    do_test_optimistic_confirmation_violation_with_or_without_tower(false);
+    do_test_lockout_violation_with_or_without_tower(false);
 }
 
 // A bit convoluted test case; but this roughly follows this test theoretical scenario:
 // Validator A, B, C have 31, 36, 33 % of stake respectively. Leader schedule is split, first half
-// of the test B is always leader, second half C is. Additionally we have a non voting validator D with 0
-// stake to propagate gossip info.
+// of the test B is always leader, second half C is.
 //
 // Step 1: Kill C, only A, B and D should be running
```

Comment:
We don't have a D now. By the way, why did we need D before but not anymore?

Reply:
It's been a while, but I think this comment is a clue: https://github.com/anza-xyz/agave/blob/41ae0a55298d9489004899943fc496c45f687c6f/local-cluster/tests/local_cluster.rs#L3166

I think it comes from a time before we added this change: https://github.com/anza-xyz/agave/blob/41ae0a55298d9489004899943fc496c45f687c6f/local-cluster/tests/local_cluster.rs#L3196-L3199 to prevent `C` from making blocks before the base slot forking point.

and when we relied on waiting for validators `A` and `C` to vote on and OC some slot > next_slot_on_a, where that slot could have been produced by validator `C`

In this scenario, in order for `C` to reliably restart and make another fork, I think it would have needed to pass a propagated check, which implies seeing the vote from A in gossip, which is no longer necessary